lux/lib/lux/prisms/discord/channel_prism.ex

```diff
@@ -0,0 +1,470 @@
+  """
+  @impl true
+  def handler(input, ctx) do
+    with {:ok, request} <- prepare_request(input),
+         {:ok, response} <- execute_request(request) do
+      format_response(response)
+    end
+  end
+
+  @doc """
+  Prepares the API request based on validated parameters.
+  """
+  @spec prepare_request(map()) :: {:ok, request_params()} | {:error, map()}
+  def prepare_request(params) do
+    try do
+      request = build_request(params)
+      {:ok, request}
+    rescue
+      e in RuntimeError ->
+        Logger.error("Error preparing request: #{inspect(e)}")
+        {:error, %{type: "preparation_error", message: e.message}}
+    end
+  end
+
+  @doc """
+  Executes the API request asynchronously with timeout handling.
+  """
+  @spec execute_request(request_params()) :: api_response()
+  def execute_request(request) do
+    task = Task.async(fn ->
+      try do
+        make_request(request)
+      rescue
+        e in Req.Error ->
+          {:error, %{type: "request_error", message: e.message}}
+      end
+    end)
+
+    case Task.yield(task, @timeout) do
+      {:ok, result} -> result
+      nil ->
+        Task.shutdown(task)
+        {:error, %{type: "timeout", message: "Request timed out"}}
+    end
+  end
+
+  # Private functions
+
+  @spec make_request(request_params()) :: api_response()
+  defp make_request(%{method: method, url: url, body: body}) do
+    headers = [
+      {"Authorization", "Bot #{Application.fetch_env!(:lux, :discord_token)}"},
+      {"Content-Type", "application/json"}
+    ]
+
+    response = case method do
+      :get -> Req.get!(url, headers: headers)
+      :post -> Req.post!(url, json: body, headers: headers)
+      :put -> Req.put!(url, json: body, headers: headers)
+      :patch -> Req.patch!(url, json: body, headers: headers)
+      :delete -> Req.delete!(url, headers: headers)
+    end
+
+    handle_response(response)
+  end
+
+  @spec handle_response(Req.Response.t()) :: api_response()
+  defp handle_response(response) do
+    case response do
+      %{status: status, body: body} when status in 200..299 ->
+        {:ok, body}
+      %{status: 429, headers: headers} ->
+        retry_after = headers["retry-after"] || "5"
+        {:error, %{
+          type: "rate_limit",
+          message: "Rate limited",
+          retry_after: String.to_integer(retry_after)
+        }}
+      %{status: status, body: %{"code" => code, "message" => message}} ->
+        {:error, %{
+          type: "discord_api_error",
+          code: code,
+          message: message,
+          status: status
+        }}
+      _ ->
+        {:error, %{
+          type: "unknown_error",
+          message: "An unexpected error occurred",
+          status: response.status
+        }}
+    end
+  end
+
+  @spec format_response(map() | {:error, map()}) :: {:ok, map()} | {:error, map()}
+  defp format_response({:error, _} = error), do: error
```

Comment:
Formatting also sounds like something that you might want to standardize and reuse as you build more prisms.